src/Rules/DateTimeInstantiationRule.php

```diff
@@ -0,0 +1,72 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules;
+
+use DateTime;
+use DateTimeImmutable;
+use PhpParser\Node;
+use PhpParser\Node\Expr\New_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Type\Constant\ConstantStringType;
+
+/**
+ * @implements \PHPStan\Rules\Rule<\PhpParser\Node\Expr\New_>
+ */
+class DateTimeInstantiationRule implements \PHPStan\Rules\Rule
+{
+
+	public function getNodeType(): string
+	{
+		return New_::class;
+	}
+
+	/**
+	 * @param New_ $node
+	 */
+	public function processNode(Node $node, Scope $scope): array
+	{
+		if (
+			!($node->class instanceof \PhpParser\Node\Name)
+			|| \count($node->args) === 0
+			|| (
+				strtolower((string) $node->class) !== strtolower(DateTime::class)
+				&& strtolower((string) $node->class) !== strtolower(DateTimeImmutable::class)
```

Comment:
1) This unnecessarily calls `strtolower((string) $node->class)` twice.
2) You can hardcode `datetime` instead.